### PR TITLE
Jacob/bugfix/heroku deployment failure

### PR DESCRIPTION
**Pull request by Goldenaries658**
Large change so comb through this one closely:

* Changed some node modules from dev to normal and vice versa
* Added some comments through
  - please note in review if there's anything that's not clear so I can comment further

# Testing
* Read code line by line.
* Connect to JAWSDB from local workbench
* Test deployment
* Ensure that db is updating when being used on heroku

**If anything fails please let me know asap so I can check logs**

## express-session
* Used to create an active session for passport to use.
  - This uses cookies so may be worth looking into converting for session storage but that's for another day.
* Works by storing a session key in the DB that relates to the user that is logged in.
  - Have a play with logging in on the deployment and checking the changes in the DB that this causes
  - This is only usable in the dev environment, need to use an alternative storage for production. so using:

## express-mysql-session
* This simply handles access to the db.
  - Has it's own setting for session expiry along with passport, they work in tandem in a weird way.
  - Doesn't accept a URL for db so tried creating a function that parsed it, which both worked and didn't.
  - Opted to store values separately in env variables on heroku and access them that way.

### CODE REVIEW

File: models/index.js [hunk 1]
```diff
@@ -17,20 +20,34 @@ const {
   DB: dbTitle,
   DB_USER: dbUser,
   DB_PASS: dbPass,
-  DB_HOST: dbHost,
-  JAWSDB_URL: dbJawsUrl
+  DB_HOST: dbHost
 } = process.env;
 
+// Setting options for db connections
```
Comment (JoeDodgson) [inline models/index.js:26]: Can this comment be changed to state 'Setting options for db connections for use in session storage'
Just so it's clear that it's used in express-mysql-session

File: models/index.js [hunk 2]
```diff
@@ -17,20 +20,34 @@ const {
...
+const sequelize = new Sequelize(dbTitle, dbUser, dbPass, {
   host: dbHost,
   dialect: 'mysql'
 });
 
-// const sequelize = config.use_env_variable
-//   ? new Sequelize(process.env[config.use_env_variable])
-//   : new Sequelize(config.database, config.username, config.password, config);
+// Starting session storage
+const mysqlStore = new MySQLStore(options);
```
Comment (JoeDodgson) [inline models/index.js:48]: How is session data consumed in passport?

File: server.js [hunk 3]
```diff
@@ -5,18 +5,25 @@ const session = require('express-session');
 const passport = require('./config/passport');
 
 // Setting up port and requiring models for syncing
-const PORT = process.env.PORT || 8080;
 const db = require('./models');
+const PORT = process.env.PORT || 8080;
+// const MySQLStore = require('express-mysql-session')(session);
```
Comment (JoeDodgson) [inline server.js:10]: Can this be removed since we assign a mysqlStore property in the db object?

File: models/index.js (hunk 1 above)
Comment (Goldenaries658) [reply models/index.js:26]: Done

File: models/index.js (hunk 2 above)
Comment (Goldenaries658) [reply models/index.js:48]: Passport accesses the data column in the sessions table created by MySQLStore, added a comment to explain this

File: server.js (hunk 3 above)
Comment (Goldenaries658) [reply server.js:10]: Done

**Pull request merged**